impr(input-history): Change style for incomplete last word in input history (@Leonabcd123)

Leonabcd123 opened the pull request:
### Description

Added a new style named "incomplete" that mirrors the error style, but uses yellow for the color. Words are marked in this style when they don't match the correct word exactly, but share the same prefix.

Implements #6690

Miodec commented:
The current code triggers for every word, but it should only trigger on the last:
<img width="587" height="226" alt="image" src="https://github.com/user-attachments/assets/8fd0ada5-de33-4ac1-947d-68e52da0521c" />

Also, please extract some of the checks to separate consts to make it easier to read.

Also also, when youre done, make sure to request a review from me so i know its ready to check.

Leonabcd123 commented:
> Also also, when youre done, make sure to request a review from me so i know its ready to check.

It seems that I can't request a review from you. The GitHub docs say I either have to be an owner or a collaborator with write access. Anyways, I added a check to make sure we're on the last word and I extracted everything to consts.

Miodec commented:
> > Also also, when youre done, make sure to request a review from me so i know its ready to check.
> 
> It seems that I can't request a review from you. The GitHub docs say I either have to be an owner or a collaborator with write access. Anyways, I added a check to make sure we're on the last word and I extracted everything to consts.

You should be able to do it, you did it fine on the other PR 
<img width="605" height="97" alt="image" src="https://github.com/user-attachments/assets/2333c4a3-6ff6-49af-894b-b16829a00751" />
